src/main/java/teammates/ui/webapi/action/GetHasResponsesAction.java

```diff
@@ -0,0 +1,62 @@
+package teammates.ui.webapi.action;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.exception.InvalidHttpParameterException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.HasResponsesData;
+
+/**
+ * Checks whether a session has responses.
+ */
+public class GetHasResponsesAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Anyone can check whether a course has responses
+
+        String questionId = getRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID);
+        if (questionId != null) {
+            FeedbackQuestionAttributes feedbackQuestionAttributes = logic.getFeedbackQuestion(questionId);
+            FeedbackSessionAttributes feedbackSession = logic.getFeedbackSession(
+                    feedbackQuestionAttributes.getFeedbackSessionName(),
+                    feedbackQuestionAttributes.getCourseId());
+
+            gateKeeper.verifyAccessible(
+                    logic.getInstructorForGoogleId(feedbackQuestionAttributes.getCourseId(), userInfo.getId()),
+                    feedbackSession);
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+        if (courseId != null) {
+            if (logic.getCourse(courseId) == null) {
+                return new JsonResult("No course with id: " + courseId, HttpStatus.SC_NOT_FOUND);
+            }
+
+            boolean hasResponses = logic.hasResponsesForCourse(courseId);
+            return new JsonResult(new HasResponsesData(hasResponses));
+        }
+
+        String feedbackQuestionID = getRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID);
```

Comment:
At this point, you can also do `getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID)` and do the logic below without the extra null check. Since you are throwing an exception below anyway when both are null, this should have the same result.

Reply:
Updated. This does mean that the error message given is slightly less accurate though - since it says missing parameter Feedback question id, which is not necessarily right.